packages/ra-data-json-server/src/index.ts

```diff
@@ -35,14 +35,17 @@ import { fetchUtils, DataProvider } from 'ra-core';
  */
 export default (apiUrl, httpClient = fetchUtils.fetchJson): DataProvider => ({
     getList: (resource, params) => {
-        const { page, perPage } = params.pagination;
-        const { field, order } = params.sort;
+        const { page, perPage } = params.pagination || {};
```

Comment:
We should have to do the same for `getManyReference`, except the Params types for the two methods don't match... params are optional in `getList` and required for `getManyReference`.